static/src/js/containers/SavedProjectsContainer/__tests__/SavedProjectsContainer.test.js

```diff
@@ -1,103 +1,259 @@
 
-    expect(spy).toBeCalledTimes(1)
+    expect(spy).toHaveBeenCalledTimes(1)
+    expect(spy).toHaveBeenCalledWith('path')
   })
 
-  test('onChangePath calls actions.changePath', () => {
+  test('dispatchHandleError calls actions.handleError', () => {
     const dispatch = jest.fn()
-    const spy = jest.spyOn(actions, 'changePath')
+    const spy = jest.spyOn(actions, 'handleError')
+    const errorConfig = { error: 'test-error' }
 
-    mapDispatchToProps(dispatch).onChangePath('path')
+    mapDispatchToProps(dispatch).dispatchHandleError(errorConfig)
 
-    expect(spy).toBeCalledTimes(1)
-    expect(spy).toBeCalledWith('path')
+    expect(spy).toHaveBeenCalledTimes(1)
+    expect(spy).toHaveBeenCalledWith(errorConfig)
   })
 })
 
 describe('mapStateToProps', () => {
   test('returns the correct state', () => {
+    getEarthdataEnvironment.mockReturnValue('prod-env')
     const store = {
-      savedProjects: {
-        projects: {},
-        isLoading: false,
-        isLoaded: false
-      }
-    }
-
-    const expectedState = {
-      savedProjects: {},
-      savedProjectsIsLoading: false,
-      savedProjectsIsLoaded: false
+      authToken: 'mock-token'
     }
 
-    expect(mapStateToProps(store)).toEqual(expectedState)
+    expect(mapStateToProps(store)).toEqual({
+      authToken: 'mock-token',
+      earthdataEnvironment: 'prod-env'
+    })
   })
 })
 
-describe('SavedProjectsContainer component', () => {
-  describe('when passed the correct props', () => {
-    test('renders a table when a retrieval exists with one collection that has no title', () => {
-      const { enzymeWrapper } = setup({
-        savedProjects: [
-          {
-            id: '4517239960',
-            name: 'project 1',
-            path: '/search?p=!C123456-EDSC',
-            created_at: '2019-09-05 00:00:00.000'
-          },
-          {
-            id: '2057964173',
-            name: 'project 3',
-            path: '/search?p=!C123456-EDSC',
-            created_at: '2019-09-05 00:00:00.000'
-          }
-        ],
-        savedProjectsIsLoading: false,
-        savedProjectsIsLoaded: true,
-        onChangePath: jest.fn(),
-        onDeleteSavedProject: jest.fn(),
-        onFetchSavedProjects: jest.fn()
+describe('SavedProjectsContainer', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    SavedProjects.mockClear()
+    nock.cleanAll()
+    nock.disableNetConnect()
+  })
+
+  describe('When the component mounts with a valid authToken', () => {
+    test('fetches the projects and renders SavedProjects with projects', async () => {
+      const mockProjects = [
+        {
+          id: '123',
+          name: 'Project 1',
+          path: '/search?p=!C123',
+          created_at: '2022-01-01T00:00:00.000Z'
+        }
+      ]
+
+      nock(/localhost/)
+        .get(/projects/)
+        .reply(200, mockProjects)
+
+      setup()
+
+      await waitFor(() => {
+        const callWithLoadedData = SavedProjects.mock.calls.find(
```

Comment:
We don't really do this in other places could we reproduce this kind of test but, using the `toHaveBeenCalledWith` and `toHaveBeenCalledTimes` apis

Reply:
+1